### DISCUSSION TITLE
How to prevent a custom model to answer questions out of its knowlege base ?

### DISCUSSION THREAD

**Discussion by YoussefAphp** [open-webui/open-webui#5519] [General]
Hi, i have implemented a RAG using open web ui, my problem is how to prevent the custom model to answer questions out of its knowledge base.

llm model : qwen2:7b
prompt which I'm using:
`You are  an IT Support Assistant Bot, focused on providing users with IT support based on the content from knowledge base. Stay in character and maintain your focus on IT support, avoiding unrelated activities such as creative writing or engaging in non-IT discussions.

If you cannot find relevant information in the knowledge base or if the user asks non-related questions that are not part of the knowledge base, do not attempt to answer and inform the user that you are unable to assist and print text "I HAVE NNO IDEA" at the end.
`
is there a tips or solution please to prevent the custom model to answer questions out of its based knowlege please ?

by the way, sometimes it works and the model answer "I HAVE NNO IDEA" and sometimes no.

**Comment by adjiap** [comment]
So, with my first system prompt (codename: Summary Sam), that's supposed to only summarize a content of a documentation, I was able to do that...but this is only after I had ask my other custom model some other stuff...

![image](https://github.com/user-attachments/assets/9d2215e0-4c0a-42a2-b8e4-77d2db043ffb)

But when I use the chat from scratch, it suddenly is able to answer (which I don't want......

![image](https://github.com/user-attachments/assets/1758a644-3bc5-4629-bc41-8fa59f010eb8)

My system prompt is fairly long, but you can look it up, [here](https://github.com/adjiap/local_ollama_powershell_setup/blob/main/system_prompts/summary_sam.md). Maybe you can see something I don't
